shared/actions/fs-platform-specific.desktop.js

```diff
@@ -102,10 +108,153 @@ function _open(openPath: string): Promise<*> {
+  // If our kextStarted status changed, finish KBFS install
+  if (status.kextStarted && prevStatus && !prevStatus.kextStarted) {
+    return Saga.call(installKBFSSaga)
+  }
+}
+
+export function* fuseStatusSaga(): Saga.SagaGenerator<any, any> {
+  const state: TypedState = yield Saga.select()
+  const prevStatus = state.favorite.fuseStatus
+
+  let status = yield Saga.call(RPCTypes.installFuseStatusRpcPromise, {bundleVersion: ''})
+  if (isWindows && status.installStatus !== 4) {
+    // Check if another Dokan we didn't install mounted the filesystem
+    const kbfsMount = yield Saga.call(RPCTypes.kbfsMountGetCurrentMountDirRpcPromise)
+    if (kbfsMount && fs.existsSync(kbfsMount)) {
+      status.installStatus = 4 // installed
+      status.installAction = 1 // none
+      status.kextStarted = true
+    }
   }
+  yield Saga.put(FsGen.createFuseStatusResult({prevStatus, status}))
+}
+
+export function* installFuseSaga(): Saga.SagaGenerator<any, any> {
+  const result: RPCTypes.InstallResult = yield Saga.call(RPCTypes.installInstallFuseRpcPromise)
+  const fuseResults =
+    result && result.componentResults ? result.componentResults.filter(c => c.name === 'fuse') : []
+  const kextPermissionError =
+    fuseResults.length > 0 && fuseResults[0].exitCode === ExitCodeFuseKextPermissionError
+
+  if (kextPermissionError) {
+    // Add a small delay here, since on 10.13 the OS will be a little laggy
+    // when showing a kext permission error.
+    yield delay(1e3)
+  }
+
+  yield Saga.put(FsGen.createInstallFuseResult({kextPermissionError}))
+  yield Saga.put(FsGen.createFuseStatus())
+  yield Saga.put(FsGen.createSetFlags({fuseInstalling: false}))
+}
+
+export function uninstallKBFSConfirmSaga(action: FsGen.UninstallKBFSConfirmPayload) {
+  const dialog = Electron.dialog || Electron.remote.dialog
+  dialog.showMessageBox(
+    {
+      buttons: ['Remove & Restart', 'Cancel'],
+      detail: `Are you sure you want to remove Keybase from ${fileUIName} and restart the app?`,
+      message: `Remove Keybase from ${fileUIName}`,
+      type: 'question',
+    },
+    resp => {
+      if (resp === 0) {
+        return action.payload.onSuccess()
+      }
+    }
+  )
+}
+
+export function uninstallKBFSSaga() {
+  return Saga.call(RPCTypes.installUninstallKBFSRpcPromise)
+}
+
+export function uninstallKBFSSagaSuccess(result: RPCTypes.UninstallResult) {
+  // Restart since we had to uninstall KBFS and it's needed by the service (for chat)
+  const app = Electron.remote.app
+  app.relaunch()
+  app.exit(0)
+}
+
+// Invoking the cached installer package has to happen from the topmost process
+// or it won't be visible to the user. The service also does this to support command line
+// operations.
+function installCachedDokan(): Promise<*> {
+  return new Promise((resolve, reject) => {
+    // use the action logger so it has a chance of making it into the upload
```

Comment:
i think this comment is now outdated. just use logger.log also

Reply:
`logger.log` doesn't exist

Reply:
Should I use `debug` or `info`?

Reply:
info